samples/web/content/apprtc/apprtc.py

```diff
@@ -495,16 +542,136 @@ def write_response(self, result, params, messages):
+      self.report_error(constants.RESPONSE_INVALID_ROOM)
+      return
+
+    allowed_gcm_ids = map(lambda x: x.gcm_id, callee_records)
+    allowed_gcm_ids.append(caller_gcm_id)
+
+    result = add_client_to_room(
+        self.request.host_url,
+        room_id,
+        caller_gcm_id,
+        False,
+        Room.TYPE_DIRECT,
+        allowed_gcm_ids)
+    if result['error'] is not None:
+      logging.info('Error adding client to room: ' + result['error'] + \
+          ', room_state=' + result['room_state'])
+      self.write_response(result['error'], {}, [])
+      return
+    
+    # TODO (chuckhays): Initiate message to verified gcm ids to ring.
+    
+    self.write_room_parameters(
+        room_id, caller_id, result['messages'], result['is_initiator'])
+      
+    logging.info('User ' + caller_id + ' initiated call to ' + callee_id + \
+        ' from gcmId ' + caller_gcm_id)
+    logging.info('User ' + caller_id + ' joined room ' + room_id)
+    logging.info('Room ' + room_id + ' has state ' + result['room_state'])
+
+  def handle_accept(self, msg, room_id):
+    """Handles a callee accepting a call request."""
+    if not util.has_msg_field(msg, PARAM_CALLEE_GCM_ID, basestring):
+      self.report_error(constants.RESPONSE_INVALID_ARGUMENT)
+      return
+
+    callee_gcm_id = msg[PARAM_CALLEE_GCM_ID]
+    # Look up and validate callee by gcm id.
+    # TODO (chuckhays): Once registration is enabled, turn on restriction
+    # to only return verified records.
+    callee_records = gcmrecord.GCMRecord.get_by_gcm_id(callee_gcm_id, False)
+    if len(callee_records) < 1:
+      self.report_error(constants.RESPONSE_INVALID_CALLEE)
+      return
+
+    callee_id = callee_records[0].user_id
+    
+    if not has_room(self.request.host_url, room_id):
+      self.report_error(constants.RESPONSE_INVALID_ROOM)
+      return
+
+    result = add_client_to_room(
+        self.request.host_url,
+        room_id,
+        callee_gcm_id,
+        False,
+        Room.TYPE_DIRECT)
+    if result['error'] is not None:
+      logging.info('Error adding client to room: ' + result['error'] + \
+          ', room_state=' + result['room_state'])
+      self.write_response(result['error'], {}, [])
+      return
+    
+    # TODO (chuckhays): Send message to ringing clients to stop ringing.
+    
+    self.write_room_parameters(
+        room_id, callee_id, result['messages'], result['is_initiator'])
+      
+    logging.info('User ' + callee_id + ' accepted call ' + \
+        'from gcmId ' + callee_gcm_id)
+    logging.info('User ' + callee_id + ' joined room ' + room_id)
+    logging.info('Room ' + room_id + ' has state ' + result['room_state'])
 
   def post(self, room_id):
+    # Check request body to determine what action to take.
+    msg = util.get_message_from_json(self.request.body)
+    if util.has_msg_field(msg, PARAM_ACTION, basestring):
+      action = msg[PARAM_ACTION]
+      if action == constants.ACTION_CALL:
```

Comment:
Sorry I didn't make it clear: I think it's better to keep the join related constants in one place, either in constants, or at the beginning of this file. Either way is fine for me.